File: lib/models/entities/tombstone_entity.dart
/// Repräsentiert einen Löschmarker ("Tombstone") für Tresoreinträge.
/// Diese Entität speichert die UUIDs von gelöschten Objekten, um die Synchronisation
/// von Löschvorgängen über mehrere Geräte hinweg zu ermöglichen.
///
/// **Funktionsweise:**
/// Wenn ein Eintrag lokal gelöscht wird, wird hier ein Grabstein hinterlassen.
/// Beim nächsten Synchronisationsvorgang meldet der Client dem Server:
/// "Eintrag mit UUID X wurde gelöscht".
class TombstoneEntity {
  /// Die interne ID (Auto-Increment in der Datenbank).
  /// Nullable für neue Einträge, bevor sie in die Datenbank geschrieben werden.
  final int? id;

  /// Die globale ID des gelöschten Eintrags (Universally Unique Identifier v4).
  final String entryUuid;

  /// Zeitpunkt (UTC) der Löschung.
  final DateTime deletedAt;

  TombstoneEntity({this.id, required this.entryUuid, required this.deletedAt});

  /// Konvertiert eine [TombstoneEntity] in eine Map (z.B. für SQLite oder JSON).
  Map<String, dynamic> toMap() {
    return {'id': id, 'entry_uuid': entryUuid, 'deleted_at': deletedAt.toIso8601String()};
  }

  /// Erstellt ein [TombstoneEntity] Objekt aus einer Map.
  factory TombstoneEntity.fromMap(Map<String, dynamic> map) {
    return TombstoneEntity(
      id: map['id'] as int?,
      entryUuid: map['entry_uuid'] as String,
      deletedAt: DateTime.parse(map['deleted_at'] as String).toUtc(),
    );
  }
}

File: old/Privault.Code/Models/Entities/TombstoneEntity.cs
using SQLite;

// ReSharper disable PropertyCanBeMadeInitOnly.Global

namespace Privault.Core.Models.Entities;

/// <summary>
/// Repräsentiert einen Löschmarker ("Tombstone") für Tresoreinträge.
/// Diese Entität speichert die UUIDs von gelöschten Objekten, um die Synchronisation 
/// von Löschvorgängen über mehrere Geräte hinweg zu ermöglichen.
/// <para>
/// <b>Funktionsweise:</b>
/// Wenn ein Eintrag lokal gelöscht wird, wird hier ein Grabstein hinterlassen. Beim nächsten 
/// Synchronisationsvorgang meldet der Client dem Server: "Eintrag mit UUID X wurde gelöscht".
/// </para>
/// </summary>
[Table("tombstones")]
public class TombstoneEntity
{
    /// <summary>
    /// Die interne ID (Auto-Increment).
    /// Dient als Primärschlüssel für den Object-Relational Mapper (ORM).
    /// </summary>
    [PrimaryKey, AutoIncrement]
    [Column("id")]
    public int Id { get; set; }
    
    /// <summary>
    /// Die globale ID des gelöschten Eintrags (Universally Unique Identifier v4).
    /// </summary>
    [Indexed(Name = "uk_tombstones_entry_uuid", Unique = true)] 
    [Column("entry_uuid")]
    public string EntryUuid { get; set; } = string.Empty;

    private DateTime _deletedAt = DateTime.SpecifyKind(new DateTime(1970, 1, 1), DateTimeKind.Utc);

    /// <summary>
    /// Zeitpunkt (UTC) der Löschung.
    /// </summary>
    [Indexed(Name = "idx_tombstones_deleted_at")]
    [Column("deleted_at")]
    public DateTime DeletedAt
    {
        get => _deletedAt;
        set => _deletedAt = value.Kind == DateTimeKind.Unspecified ? DateTime.SpecifyKind(value, DateTimeKind.Utc) : value.ToUniversalTime();
    }
}